materialize-bigquery/binding.go

```diff
@@ -1,225 +1,319 @@
-
-		colType, err := baseTypeMapper.GetColumnType(col)
-		if err != nil {
-			return nil, err
-		}
+	return nil
+}
 
-		b.store.extDataConfig.Schema = append(b.store.extDataConfig.Schema, &bigquery.FieldSchema{
-			Name:     identifierSanitizer(col.Name), // Sanitized name without quoting required
-			Repeated: false,
-			Required: col.PrimaryKey,
-			Type:     bigquery.FieldType(colType.SQLType),
-		})
+// Delete the object associated to this Writer. Once deleted, the file
+// cannot be recovered. This is a destructive operation that should only
+// happen once the underlying data is written to bigquery. Otherwise, it could result
+// in data loss.
+func (w *writer) Destroy(ctx context.Context) error {
+	return w.object.Delete(ctx)
+}
 
-		colIdentifiers = append(colIdentifiers, col.Identifier)
-		rColIdentifiers = append(rColIdentifiers, fmt.Sprintf("r.%s", col.Identifier))
+func randomString() (string, error) {
+	id, err := uuid.NewRandom()
+	if err != nil {
+		return "", fmt.Errorf("generating UUID: %w", err)
 	}
 
-	// If a field is selected whose document pointer is the root document, then Store()
-	// will need to include the root document in the set of values being written.
-	b.store.hasRootDocument = spec.FieldSelection.Document != ""
+	return id.String(), nil
+}
+
+// Generates the sqlQueries that reflects the specification for the provided materialization
+// binding.
+//
+func generateSQLQueries(externalTableAlias string, br bindingResource, bindingSpec *pf.MaterializationSpec_Binding) *sqlQueries {
+	var loadQuery, writeQuery string
+	var columns []string
+	var rColumns []string
+
+	var loadPredicates []string
+	for i, value := range bindingSpec.FieldSelection.Keys {
+		loadPredicates = append(loadPredicates, fmt.Sprintf("%s = '%s'", bindingSpec.FieldSelection.Keys[i], value))
+	}
 
-	b.store.tempTableName = fmt.Sprintf("%s_store_%d", tempTableNamePrefix, bindingPos)
+	for _, key := range bindingSpec.FieldSelection.AllFields() {
+		columns = append(columns, key)
+		rColumns = append(rColumns, fmt.Sprintf("r.%s", key))
+	}
 
-	if spec.DeltaUpdates {
-		// Store everything (do not merge/upsert).
-		b.store.sql = fmt.Sprintf(`
+	if bindingSpec.DeltaUpdates {
+		loadQuery = "ASSERT false AS 'Load queries should never be executed in Delta Updates mode.';"
+		writeQuery = fmt.Sprintf(`
 		INSERT INTO %s (%s)
 		SELECT %s FROM %s
 		;`,
-			tableDef.Identifier,
-			strings.Join(colIdentifiers, ", "),
-			strings.Join(colIdentifiers, ", "),
-			b.store.tempTableName,
+			br.Table,
+			strings.Join(columns, ", "),
+			strings.Join(rColumns, ", "),
+			br.Table,
 		)
+
 	} else {
-		// Updates for all columns minus the primary keys plus the document field.
-		var lrUpdates []string
-		for _, colName := range append(spec.FieldSelection.Values, spec.FieldSelection.Document) {
-			var col = tableDef.GetColumn(colName)
-			lrUpdates = append(lrUpdates, fmt.Sprintf("l.%s = r.%s", col.Identifier, col.Identifier))
+		loadQuery = fmt.Sprintf(`
```

Comment:
One thing to look into here is quoting of column and table names. I think the old sql driver had some functionality for quoting identifiers that needed it, depending on which database we're talking to, and we probably want the same behavior here, just with [bigquery-specific rules](https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical). As written, I think this would fail if a column name happened to be a reserved word or start with a number.